workbench-core/environments/src/services/environmentService.test.ts

```diff
@@ -1042,5 +1042,104 @@ describe('EnvironmentService', () => {
+
+      // Write data to DDB
+      ddbMock.on(TransactWriteItemsCommand).resolves({});
+
+      // Get environment from DDB
+      const metaData = [datasetItem, envTypeConfigItem, projItem];
+      const envWithMetadata = [env, ...metaData];
+      const queryItemResponse: QueryCommandOutput = {
+        Items: envWithMetadata.map((item) => {
+          return marshall(item);
+        }),
+        $metadata: {}
+      };
+      ddbMock.on(QueryCommand).resolves(queryItemResponse);
+
+      // OPERATE && CHECK
+      await expect(envService.createEnvironment(createEnvReq, authenticateUser)).rejects.toThrow(
+        'envTypeId envType-123 with envTypeConfigId envTypeConfig-123 does not exist'
+      );
+    });
+    test('failed because Project does not exist', async () => {
+      // BUILD
+      // Get env metadata
+      const batchItems = {
+        Responses: {
+          [TABLE_NAME]: [
+            marshall({ ...envTypeConfigItem, resourceType: 'envTypeConfig' }),
+            marshall({ ...datasetItem, resourceType: 'dataset' })
+          ] // Order is important
+        }
+      };
+      // @ts-ignore
+      ddbMock.on(BatchGetItemCommand).resolves(batchItems);
+
+      // Write data to DDB
+      ddbMock.on(TransactWriteItemsCommand).resolves({});
+
+      // Get environment from DDB
+      const metaData = [datasetItem, envTypeConfigItem, projItem];
+      const envWithMetadata = [env, ...metaData];
+      const queryItemResponse: QueryCommandOutput = {
+        Items: envWithMetadata.map((item) => {
+          return marshall(item);
+        }),
+        $metadata: {}
+      };
+      ddbMock.on(QueryCommand).resolves(queryItemResponse);
+
+      // OPERATE && CHECK
+      await expect(envService.createEnvironment(createEnvReq, authenticateUser)).rejects.toThrow(
+        'projectId proj-123 does not exist'
+      );
+    });
+    test('failed because Dataset does not exist', async () => {
+      // BUILD
+      // Get env metadata
+      const batchItems = {
+        Responses: {
+          [TABLE_NAME]: [
+            marshall({ ...envTypeConfigItem, resourceType: 'envTypeConfig' }),
+            marshall({ ...projItem, resourceType: 'project' })
+          ] // Order is important
+        }
+      };
+      // @ts-ignore
+      ddbMock.on(BatchGetItemCommand).resolves(batchItems);
+
+      // Write data to DDB
+      ddbMock.on(TransactWriteItemsCommand).resolves({});
+
+      // Get environment from DDB
+      const metaData = [datasetItem, envTypeConfigItem, projItem];
+      const envWithMetadata = [env, ...metaData];
+      const queryItemResponse: QueryCommandOutput = {
+        Items: envWithMetadata.map((item) => {
+          return marshall(item);
+        }),
+        $metadata: {}
+      };
+      ddbMock.on(QueryCommand).resolves(queryItemResponse);
+
+      // OPERATE && CHECK
+      await expect(envService.createEnvironment(createEnvReq, authenticateUser)).rejects.toThrow(
```

Comment:
another non-blocker but something to note:  I think `toThrow` doesn't check the error code, just the error message. Might be worth investigating if Jest has a Boom extension, otherwise we can write it ourselves. thoughts?

Reply:
updated to expect specific boom error

Reply:
If you change this to `Boom.somethingElse` but leave the message the same does the test fail? I believe thats the behavior i ran into in the past; the message was the only thing being checked, not the error type.